mlflow/utils/autologging_utils.py

```diff
@@ -804,6 +804,7 @@ def with_managed_run(autologging_integration, patch_function, tags=None):
     """
 
     def create_managed_run():
+        run_id_env_var_exists = mlflow.tracking._RUN_ID_ENV_VAR in os.environ
         managed_run = mlflow.start_run(tags=tags)
```

Comment:
Note that `mlflow.start_run` deletes `MLFLOW_RUN_ID` right after extracting a run_id from it:

https://github.com/mlflow/mlflow/blob/4ba469ea5d67ca327c23a6e57d1fc2d2256c6f1c/mlflow/tracking/fluent.py#L178-L180